modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/SnapshotCompatibilityTest.java

```diff
@@ -0,0 +1,452 @@
+
+        return cfg;
+    }
+
+    /** */
+    private static String customSnapshotPath(String relativePath, boolean forSnapshotTake) throws IgniteCheckedException {
+        return U.resolveWorkDirectory(U.defaultWorkDirectory(), relativePath, forSnapshotTake).getAbsolutePath();
+    }
+
+    /** */
+    private static String calcValue(String cacheName, int key) {
+        return cacheName + "-organization-" + key;
+    }
+
+    /**
+     * Configuration closure both for old and current Ignite version.
+     */
+    private static class ConfigurationClosure implements IgniteInClosure<IgniteConfiguration> {
+        /** */
+        private final String consistentId;
+
+        /** */
+        private final boolean incrementalSnp;
+
+        /** */
+        private final boolean customSnpPath;
+
+        /** */
+        private final boolean forSnapshotTake;
+
+        /** */
+        private final CacheGroupInfo cacheGrpInfo;
+
+        /** */
+        public ConfigurationClosure(
+            boolean incrementalSnp,
+            String consistentId,
+            boolean customSnpPath,
+            boolean forSnapshotTake,
+            CacheGroupInfo cacheGrpInfo
+        ) {
+            this.consistentId = consistentId;
+            this.incrementalSnp = incrementalSnp;
+            this.customSnpPath = customSnpPath;
+            this.forSnapshotTake = forSnapshotTake;
+            this.cacheGrpInfo = cacheGrpInfo;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void apply(IgniteConfiguration igniteConfiguration) {
+            DataStorageConfiguration storageCfg = new DataStorageConfiguration();
+
+            storageCfg.getDefaultDataRegionConfiguration().setPersistenceEnabled(true);
+
+            igniteConfiguration.setDataStorageConfiguration(storageCfg);
+
+            igniteConfiguration.setConsistentId(consistentId);
+
+            if (incrementalSnp)
+                storageCfg.setWalCompactionEnabled(true);
+
+            if (forSnapshotTake)
+                igniteConfiguration.setCacheConfiguration(
+                    cacheGrpInfo.cacheNamesList().stream()
+                        .map(cacheName -> new CacheConfiguration<Integer, String>(cacheName)
+                            .setGroupName(cacheGrpInfo.name())
+                            .setAffinity(new RendezvousAffinityFunction(false, 10))
+                        )
+                        .toArray(CacheConfiguration[]::new)
+                );
+
+            if (customSnpPath) {
+                try {
+                    igniteConfiguration.setSnapshotPath(customSnapshotPath(CUSTOM_SNP_RELATIVE_PATH, forSnapshotTake));
+                }
+                catch (IgniteCheckedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Post startup closure for old Ignite version.
+     */
+    private static class PostStartupClosure implements IgniteInClosure<Ignite> {
```

Comment:
Let's rename this PostStartupClosure -> CreateSnapshotClosure for better readability.